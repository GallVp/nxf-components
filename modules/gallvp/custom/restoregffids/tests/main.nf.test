nextflow_process {

    name "Test Process CUSTOM_RESTOREGFFIDS"
    script "../main.nf"
    process "CUSTOM_RESTOREGFFIDS"

    tag "modules"
    tag "modules_gallvp"
    tag "custom"
    tag "custom/restoregffids"

    test("sarscov2-genome_gff3-success") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.gff3', checkIfExists: true)
                ]
                input[1] = Channel.of('Chr1\tMT192765.1').collectFile(name: 'id_map.tsv', newLine: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert path(process.out.restored_ids_gff3.get(0).get(1)).getText().contains("Chr1") },
                { assert !path(process.out.restored_ids_gff3.get(0).get(1)).getText().contains("MT192765.1") },
                { assert snapshot(process.out.versions).match("versions") }
            )
        }

    }

    test("stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.gff3', checkIfExists: true)
                ]
                input[1] = Channel.of('Chr1\tMT192765.1').collectFile(name: 'id_map.tsv', newLine: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.restored_ids_gff3 != null },
                { assert snapshot(process.out.versions).match("versions_stub") }
            )
        }

    }

}