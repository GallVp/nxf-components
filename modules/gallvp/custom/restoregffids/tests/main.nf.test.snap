{
    "sarscov2-genome_gff3-success": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.restored.ids.gff3:md5,2c294938b9eb4e52d19e14725c1d92a9"
                    ]
                ],
                "1": [
                    "versions.yml:md5,32d31c4f1da9a3d1be013fd163e5867e"
                ],
                "restored_ids_gff3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.restored.ids.gff3:md5,2c294938b9eb4e52d19e14725c1d92a9"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,32d31c4f1da9a3d1be013fd163e5867e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2023-12-07T13:49:30.047425"
    },
    "versions_stub": {
        "content": [
            [
                "versions.yml:md5,32d31c4f1da9a3d1be013fd163e5867e"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-10T15:26:11.66528"
    },
    "versions": {
        "content": [
            [
                "versions.yml:md5,32d31c4f1da9a3d1be013fd163e5867e"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2023-12-07T13:49:30.071175"
    }
}